test-data/unit/check-modules.test

```diff
@@ -1955,3 +1957,217 @@ import c
 tmp/b.py:1: error: Cannot find module named 'c'
 main:1: error: Cannot find module named 'c'
 main:1: note: (Perhaps setting MYPYPATH or using the "--ignore-missing-imports" flag would help)
+
+[case testIndirectFromImportWithinCycle]
+import a
+[file a.py]
+from b import f
```

Comment:
Would it make sense to check that the result is independent on the order of import statements (just in case) here and/or in other tests?

Reply:
I'll add a few test cases where we reverse the shape of the import cycle relative to the main module.